### PR TITLE
fix: 2m tall voxel grid cell

### PR DESCRIPTION
**Pull request by DrCodeWizzard**

### CODE REVIEW

File: pcl_ros/cfg/VoxelGrid.cfg [hunk 1]
```diff
@@ -13,6 +13,8 @@ from dynamic_reconfigure.parameter_generator_catkin import *
 gen = ParameterGenerator ()
 # def add (self, name, paramtype, level, description, default = None, min = None, max = None, edit_method = ""):
 gen.add ("leaf_size", double_t, 0, "The size of a leaf (on x,y,z) used for downsampling.", 0.01, 0, 1.0)
+gen.add ("leaf_size_z", double_t, 0, "The size of a leaf on z used for downsampling.", 0, 0, 2.0)
```
Comment (dantwinkler) [inline pcl_ros/cfg/VoxelGrid.cfg]: Can we just change this to have x y and z?

Comment (dantwinkler) [reply pcl_ros/cfg/VoxelGrid.cfg]: And default to same size as x and y for now.

Comment (DrCodeWizzard) [reply pcl_ros/cfg/VoxelGrid.cfg]: We could... do you ever envision changing x separately from y?

File: pcl_ros/src/pcl_ros/filters/gpu/voxel_grid.cpp [hunk 2]
```diff
@@ -74,15 +74,29 @@ pcl_ros::gpu::VoxelGrid::config_callback (pcl_ros::VoxelGridConfig &config, uint
...
-        impl_.setLeafSize (leaf_size[0], leaf_size[1], leaf_size[2]);
+        NODELET_ERROR("[config_callback] All config leaf_size values are set to 0. Setting to 0.01");
+        leaf_size = {0.01, 0.01, 0.01};
+    } else if (config.leaf_size_x > 0 && config.leaf_size_y > 0 && config.leaf_size_z > 0) {
+        NODELET_WARN("pconfig_callback] All leaf values are set. Using the leaf_size_x, leaf_size_y, leaf_size_z values.");
+        leaf_size.setConstant (0);
+        leaf_size[0] = config.leaf_size_x;
+        leaf_size[1] = config.leaf_size_y;
+        leaf_size[2] = config.leaf_size_z;
+    } else
```
Comment (dantwinkler) [inline pcl_ros/src/pcl_ros/filters/gpu/voxel_grid.cpp]: else should be on another line.

**Review by dantwinkler — APPROVED** [review]

**Pull request merged**